lib/ldclient-rb/impl/data_system/fdv1.rb

```diff
@@ -0,0 +1,173 @@
+          )
+
+          # Create status provider with store wrapper
+          @data_store_status_provider = LaunchDarkly::Impl::DataStore::StatusProvider.new(
+            @store_wrapper,
+            @data_store_update_sink
+          )
+
+          # Set up data source plumbing
+          @data_source_broadcaster = LaunchDarkly::Impl::Broadcaster.new(@shared_executor, @config.logger)
+          @flag_change_broadcaster = LaunchDarkly::Impl::Broadcaster.new(@shared_executor, @config.logger)
+          @data_source_update_sink = LaunchDarkly::Impl::DataSource::UpdateSink.new(
+            @store_wrapper,
+            @data_source_broadcaster,
+            @flag_change_broadcaster
+          )
+          @data_source_status_provider = LaunchDarkly::Impl::DataSource::StatusProvider.new(
+            @data_source_broadcaster,
+            @data_source_update_sink
+          )
+
+          # Ensure v1 processors can find the sink via config for status updates
+          @config.data_source_update_sink = @data_source_update_sink
+
+          # Update processor created in start()
+          @update_processor = nil
+
+          # Diagnostic accumulator provided by client for streaming metrics
+          @diagnostic_accumulator = nil
+        end
+
+        # (see DataSystem#start)
+        def start
+          @update_processor ||= make_update_processor
+          @update_processor.start
+        end
+
+        # (see DataSystem#stop)
+        def stop
+          @update_processor&.stop
+          @shared_executor.shutdown
+        end
+
+        # (see DataSystem#store)
+        def store
+          @store_wrapper
+        end
+
+        # (see DataSystem#set_diagnostic_accumulator)
+        def set_diagnostic_accumulator(diagnostic_accumulator)
+          @diagnostic_accumulator = diagnostic_accumulator
+        end
+
+        # (see DataSystem#data_source_status_provider)
+        def data_source_status_provider
+          @data_source_status_provider
+        end
+
+        # (see DataSystem#data_store_status_provider)
+        def data_store_status_provider
+          @data_store_status_provider
+        end
+
+        # (see DataSystem#flag_change_broadcaster)
+        def flag_change_broadcaster
+          @flag_change_broadcaster
+        end
+
+        #
+        # (see DataSystem#data_availability)
+        #
+        # In LDD mode, always returns CACHED for backwards compatibility,
+        # even if the store is empty.
+        #
+        def data_availability
+          return DataAvailability::DEFAULTS if @config.offline?
+
+          # In LDD mode, always return CACHED for backwards compatibility.
+          # Even though the store might be empty (technically DEFAULTS), we maintain
+          # the existing behavior where LDD mode is assumed to have data available
+          # from the external daemon, regardless of the store's initialization state.
+          return DataAvailability::CACHED if @config.use_ldd?
```

Comment:
I actually think it should be refreshed to be compatible with FDv1. Previously, we didn't check for `use_ldd?` mode here. Instead, we would have fallen through and the null update processor, which is always initialized, would have returned true. You should be able to just remove this line.